realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -897,24 +897,139 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeNotEqual__J_3JLja
     TableQuery_StringPredicate(env, nativeQueryPtr, columnIndexes, value, caseSensitive, StringNotEqual);
 }
 
-JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeBeginsWith(
+JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeBeginsWith__J_3JLjava_lang_String_2Z(
```

Comment:
I think that `javah` does this to avoid overloading. It is basically C code, and overloading was introduced in C11. `javah` is much older than that. I have noticed a similar issue/annoyance in Android Studio 2.2 beta.